minor updates to session middleware

basvanbeek opened the pull request:
RegenerateId allows you to regenerate the sessionId to fence off session fixation attacks
Cookie storing sessionid now has session scope (removed on browser close) and sessions time out server side
SetTimeout allows to alter the session timeout in an easy way during runtime
SetHttpOnly allows configuring http/server only session cookie
SetSecure allows configuring secure https only session cookie
BaseSessionHolder and session implementations (redis/memory) now implement ResetTTL logic so session timeout counter resets to 0 on successful request, even if session data is unaltered in request handler

basvanbeek committed:
added session regeneration logic to combat session fixation attacks

basvanbeek committed:
Added ability to update timeout of session, added update of TTL of a session object if session not dirty, improved cookie to be safer according to OWasp recommendations (secure and httponly flags)

basvanbeek committed:
added some comments

basvanbeek committed:
merged with incoming upstream changes

basvanbeek committed:
added secure and httponly cookie on/off methods

basvanbeek committed:
removed MaxAge from cookie in favor of session scoped cookies

philpearl commented:
This all looks great, but I'm afraid I can't accept session scoped cookies as the only option.  For my application its much more important to keep the user signed in.

basvanbeek commented:
Sounds reasonable, what if I update the code to allow switching between session scoped cookies and time out scoped cookies where time out scoped cookies are the default?

philpearl commented:
That would be absolutely brilliant.

On Mon, 26 Jan 2015 18:34 Bas van Beek notifications@github.com wrote:

> Sounds reasonable, what if I update the code to allow switching between
> session scoped cookies and time out scoped cookies where time out scoped
> cookies are the default?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/philpearl/tt_goji_middleware/pull/5#issuecomment-71511992
> .

basvanbeek committed:
added ability to switch between persistent cookies using MaxAge and session based cookies which get deleted on browser close

basvanbeek committed:
fixed redis session test

basvanbeek committed:
renamed SetPersistentCookie to SetPersistentCookies

basvanbeek commented:
Hi Phil... I added ability to switch between persistent cookies and session cookies and updated the test suite... by default persistent Max-Age'd cookies are used as your current library does, but can be switched to session cookies using sh.SetPersistentCookies(false)

philpearl commented:
Absolutely brilliant.  Thanks!
